AsyncDisplayKit/Details/ASEnvironment.h

```diff
@@ -154,13 +156,15 @@ ASDISPLAYNODE_EXTERN_C_END
   [super setEnvironmentState:environmentState];\
   ASEnvironmentTraitCollection currentTraits = environmentState.traitCollection;\
   if (ASEnvironmentTraitCollectionIsEqualToASEnvironmentTraitCollection(currentTraits, oldTraits) == NO) {\
-    NSArray<NSArray <ASCellNode *> *> *completedNodes = [self.view.dataController completedNodes];\
-    for (NSArray *sectionArray in completedNodes) {\
-      for (ASCellNode *cellNode in sectionArray) {\
-        ASEnvironmentStatePropagateDown(cellNode, currentTraits);\
-        [cellNode setNeedsLayout];\
+    dispatch_async(dispatch_get_main_queue(), ^{\
```

Comment:
@rcancro does completedNodes currently assert that it should be called on the main thread?  If it does not but that is required, we should add that...

Because any transactions are performed on a different array, it may actually the be straightforward to add a walk and allow multithreaded access to this.  @maicki knows that code very well and should consider if this is indeed a simple change, in which case he may be able to remove this thread trampoline for us later.

Probably not a top priority, but I don't think I will have an opportunity to fully consider the implications of the trampoline, so let me know if there is any hypothesis for how it could introduce a problem.

For example, could there be edit transactions scheduled to occur on the next run loop that completely rewrite the data source and all of the nodes, meaning that we should not perform this operation? What happens if this operation is triggered from multiple threads, Is it possible that they will race in a way that cannot be controlled by Application level synchronization?